apps/backend/src/agents/persona-agent.ts

```diff
@@ -222,36 +232,43 @@ export class CompanionAgent {
           throw new Error(`Failed to build context for stream ${streamId}`)
         }
 
-        // Build system prompt with stream context
-        const systemPrompt = buildSystemPrompt(persona, context)
+        // Build system prompt with stream context and trigger info
+        const systemPrompt = buildSystemPrompt(persona, context, trigger)
+
+        // Track target stream for responses - may change if we create a thread
+        let targetStreamId = streamId
+        let threadCreated = false
+
+        // Helper to send a message, creating a thread for channel mentions on first send
+        const doSendMessage = async (msgInput: SendMessageInputWithSources): Promise<SendMessageResult> => {
+          // For channel mentions: create thread on first message
+          if (trigger === "mention" && context.streamType === StreamTypes.CHANNEL && !threadCreated) {
+            const thread = await createThread({
+              workspaceId,
+              parentStreamId: streamId,
+              parentMessageId: messageId,
+              createdBy: persona.id,
+            })
+            targetStreamId = thread.id
+            threadCreated = true
+            logger.info({ threadId: thread.id, streamId, messageId }, "Created thread for channel mention response")
+          }
+
+          const message = await createMessage({
+            workspaceId,
+            streamId: targetStreamId,
+            authorId: persona.id,
+            authorType: AuthorTypes.PERSONA,
+            content: msgInput.content,
+            sources: msgInput.sources,
+          })
+          return { messageId: message.id, content: msgInput.content }
+        }
 
         // Create callbacks for the response generator
         const callbacks: ResponseGeneratorCallbacks = {
-          // Used by send_message tool (LLM-initiated)
```

Comment:
Given that we fetch return `stream` during the pre-check, perhaps we can simplify `buildStreamContext` to take in the stream instead of looking it up?